### ISSUE TITLE
a11y: ARIA role defined inappropriately

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13812]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

a11y: issue `23`

Header Area 
- Pin/Unpin image buttons (which gets available after hovering on local and vpat-testing-downstream button) 

The aria-role="button" attribute is inappropriately used on mentioned images resulting in screen reader users to not communicate appropriate role information.

Remove aria-role="button" and provide role="button" to <i> element containing mentioned image buttons.

a11y: issue `152`

The aria-role="button" attribute is inappropriately used on <i> element containing mentioned image button resulting in screen reader users are not communicated with appropriate role information. Additionally, state is inappropriately defined for the image button using "aria-checked" attribute.

ARIA role should be appropriately used with correct semantics. Also, selected and not selected state should be appropriately announced to screen reader users.

• Remove aria-role="button" from the mentioned image button and provide role="button" to it.
• Specify aria-pressed="true" by default in the source code of mentioned button. Ensure that the value of aria-pressed attribute changes on user interaction via scripting.

Main content
- Favourite image button

**To Reproduce**

**Result**

**Expected Result**

ARIA role should be appropriately used with correct semantics.

**Screenshots**

**Additional context**

**Comment by a-arias** [comment]
TESTS
 Rancher : v2.12-80b7e22d0b44cd68bc51d5edff3d3736bcfc0553-head
UI: master https://github.com/rancher/dashboard/commit/c68a7da9da0926c669a6741b1abb16760bba283f
Fix verified ✅ 

https://github.com/user-attachments/assets/141307b2-bbc4-4be4-ba89-ab56df9543f8

<img width="1393" alt="Image" src="https://github.com/user-attachments/assets/b2e3def8-9ef4-48b8-942b-56c6135f1ab3" />
